flink-connector-kudu/src/main/java/org/apache/flink/connectors/kudu/table/KuduCatalog.java

```diff
@@ -0,0 +1,348 @@
+
+            return table;
+        } catch (KuduException e) {
+            throw new CatalogException(e);
+        }
+    }
+
+    protected Map<String, String> createTableProperties(String tableName, List<ColumnSchema> primaryKeyColumns) {
+        Map<String, String> props = new HashMap<>();
+        props.put(KUDU_MASTERS, kuduMasters);
+        String primaryKeyNames = primaryKeyColumns.stream().map(ColumnSchema::getName).collect(Collectors.joining(","));
+        props.put(KUDU_PRIMARY_KEY_COLS, primaryKeyNames);
+        props.put(KuduTableFactory.KUDU_TABLE, tableName);
+        return props;
+    }
+
+    @Override
+    public void dropTable(ObjectPath tablePath, boolean ignoreIfNotExists) throws TableNotExistException {
+        String tableName = tablePath.getObjectName();
+        try {
+            if (tableExists(tablePath)) {
+                kuduClient.deleteTable(tableName);
+            } else if (!ignoreIfNotExists) {
+                throw new TableNotExistException(getName(), tablePath);
+            }
+        } catch (KuduException e) {
+            throw new CatalogException("Could not delete table " + tableName, e);
+        }
+    }
+
+    @Override
+    public void renameTable(ObjectPath tablePath, String newTableName, boolean ignoreIfNotExists) throws TableNotExistException {
+        String tableName = tablePath.getObjectName();
+        try {
+            if (tableExists(tablePath)) {
+                kuduClient.alterTable(tableName, new AlterTableOptions().renameTable(newTableName));
+            } else if (!ignoreIfNotExists) {
+                throw new TableNotExistException(getName(), tablePath);
+            }
+        } catch (KuduException e) {
+            throw new CatalogException("Could not rename table " + tableName, e);
+        }
+    }
+
+    public void createTable(KuduTableInfo tableInfo, boolean ignoreIfExists) throws CatalogException, TableAlreadyExistException {
+        ObjectPath path = getObjectPath(tableInfo.getName());
+        if (tableExists(path)) {
+            if (ignoreIfExists) {
+                return;
+            } else {
+                throw new TableAlreadyExistException(getName(), path);
+            }
+        }
+
+        try {
+            kuduClient.createTable(tableInfo.getName(), tableInfo.getSchema(), tableInfo.getCreateTableOptions());
+        } catch (
+                KuduException e) {
+            throw new CatalogException("Could not create table " + tableInfo.getName(), e);
+        }
+    }
+
+    @Override
+    public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ignoreIfExists) throws TableAlreadyExistException {
+        Map<String, String> tableProperties = table.getProperties();
+        TableSchema tableSchema = table.getSchema();
+
+        Set<String> optionalProperties = new HashSet<>(Arrays.asList(KUDU_REPLICAS));
+        Set<String> requiredProperties = new HashSet<>(Arrays.asList(KUDU_HASH_COLS));
```

Comment:
Are range partitions supported?

Reply:
We did not implement that at this point, no.